client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -170,23 +177,24 @@ public ControllerImpl(ManagedChannelBuilder<?> channelBuilder, final ControllerI
         long traceId = LoggerHelpers.traceEnter(log, "createScope", scopeName);
 
         final CompletableFuture<CreateScopeStatus> result = this.retryConfig.runAsync(() -> {
-                RPCAsyncCallback<CreateScopeStatus> callback = new RPCAsyncCallback<>(traceId, "createScope");
-                client.createScope(ScopeInfo.newBuilder().setScope(scopeName).build(), callback);
-                return callback.getFuture();
+            RPCAsyncCallback<CreateScopeStatus> callback = new RPCAsyncCallback<>(traceId, "createScope");
+            addTagsToRequest(client, traceId, "createScope", scopeName)
```

Comment:
Can we instead wrap the client stub with a new class so that we can have a more fluent way of calling this? Concretely, I'm suggesting changes such that I can do the following:

```
client.withTag(traceId, "createScope", scopeName)
      .createScope(ScopeInfo.newBuilder().setScope(scopeName).build(), callback);
```

What do you think?

Reply:
Following your suggestion, I created an internal wrapper class named `ControllerClientTagger`
that i) abstracts the logic for attaching tags to client RPC requests, and ii) exposes only the operations that are currently supported by the tracing mechanism. This works as follows:
```
new ControllerClientTagger(client).withTag(requestId, "truncateStream", scope, stream)
                                  .truncateStream(ModelHelper.decode(scope, stream, streamCut), callback);
```

Hope this improves code clarity.

Reply:
Following your suggestion, I created an internal wrapper class named `ControllerClientTagger`
that i) abstracts the logic for attaching tags to client RPC requests, and ii) exposes only the operations that are currently supported by the tracing mechanism. This works as follows:
```
new ControllerClientTagger(client).withTag(requestId, "truncateStream", scope, stream)
                                  .truncateStream(ModelHelper.decode(scope, stream, streamCut), callback);
```

Hope this improves code clarity.

Reply:
Following your suggestion, I created an internal wrapper class named `ControllerClientTagger`
that i) abstracts the logic for attaching tags to client RPC requests, and ii) exposes only the operations that are currently supported by the tracing mechanism. This works as follows:
```
new ControllerClientTagger(client).withTag(requestId, "truncateStream", scope, stream)
                                  .truncateStream(ModelHelper.decode(scope, stream, streamCut), callback);
```

Hope this improves code clarity.